internal/blueprint/repository_customizations.go

```diff
@@ -0,0 +1,326 @@
+package blueprint
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/osbuild/osbuild-composer/internal/common"
+)
+
+type RepositoryCustomization struct {
+	Id           string   `json:"id" toml:"id"`
+	BaseURLs     []string `json:"baseurls,omitempty" toml:"baseurls,omitempty"`
+	GPGKeys      []string `json:"gpgkeys,omitempty" toml:"gpgkeys,omitempty"`
+	Metalink     string   `json:"metalink,omitempty" toml:"metalink,omitempty"`
+	Mirrorlist   string   `json:"mirrorlist,omitempty" toml:"mirrorlist,omitempty"`
+	Name         string   `json:"name,omitempty" toml:"name,omitempty"`
+	Priority     *int     `json:"priority,omitempty" toml:"priority,omitempty"`
+	Enabled      *bool    `json:"enabled,omitempty" toml:"enabled,omitempty"`
+	GPGCheck     *bool    `json:"gpgcheck,omitempty" toml:"gpgcheck,omitempty"`
+	RepoGPGCheck *bool    `json:"repo_gpgcheck,omitempty" toml:"repo_gpgcheck,omitempty"`
+	SSLVerify    bool     `json:"sslverify,omitempty" toml:"sslverify,omitempty"`
+	Filename     string   `json:"filename,omitempty" toml:"filename,omitempty"`
+}
+
+func (rc *RepositoryCustomization) UnmarshalTOML(data interface{}) error {
```

Comment:
Why do we need the custom unmarshaller?  Is it just for checking required fields and their types?  Would it be easier to check it when reading instead of needing to define our own functions?

Reply:
I think this was a suggestion from @thozza... and then we're doing the validation inside of this function rather than doing it when generating the manifest.

Reply:
I suggested this. Doing it in e.g. `GetRepositories()` would be nice, but it does not return any error, but it would not be a problem to add one. The custom unmarshaller is a bit of an overkill, I admit.

Reply:
Ah good point, I'll add it to the `GetRepositories()` function instead and return an error

Reply:
Updated this... what I don't love about it though is that I have to call this twice,
Here (to validate the repos):
https://github.com/osbuild/osbuild-composer/blob/18240dee91343c50bacf2fb1dd102fc0924b54a2/internal/distro/fedora/distro.go#L849

And here:
https://github.com/osbuild/osbuild-composer/blob/18240dee91343c50bacf2fb1dd102fc0924b54a2/internal/distro/fedora/images.go#L147-L152

It feels weird not calling the validation earlier and waiting until the second call and panicking. Would it be better to export the validation function again at that point? Or is this approach okay?

Reply:
@kingsleyzissou I think that this approach is OK. As you noted in the comment, we should not reach the second call if there was an error and if we did, it is a programming error.

Going forward, it would probably make sense to add a `Validate()` method to the `blueprint.Customizations` structure and call only that from anywhere (in image definitions, APIs, etc). Such method should then validate each specific customization using its own validator.